### PR TITLE
Fix default chat check, add serverResponse file

### PR DESCRIPTION
**Pull request by NicolasKul**
https://user-images.githubusercontent.com/43201911/199626840-c260f680-2a5a-46cc-b30f-c21426524b5f.mov

- Fix default chat check
- Add function for **Fech**

### CODE REVIEW

File: js/chat_app/src/components/serverResponse.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React, { useContext } from "react";
+
+export const serverResponse = (url, header, method = "GET", body = null) => {
```
Comment (miromyke) [inline js/chat_app/src/components/serverResponse.js]:
```suggestion
export const fetchJSON = (url, {method = "GET", body = undefined, headers = {}}) => {
```

File: js/chat_app/src/components/serverResponse.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  switch (method) {
```
Comment (miromyke) [inline js/chat_app/src/components/serverResponse.js]: `switch` creates more problems than it solves

File: js/chat_app/src/components/getServerResponse.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React, { useContext } from "react";
+
+export const getServerResponse = (
```
Comment (miromyke) [inline js/chat_app/src/components/getServerResponse.js]: duplicated file

File: js/chat_app/src/components/DashBoard.js [hunk 4]
```diff
@@ -27,6 +28,15 @@ export const DashBoard = () => {
     setOpen(false);
   };
 
+  useEffect(() => {
+    const url = `api/users/${auth?.user.id}/chats`;
+    const token = auth?.authToken;
+
+    getServerResponse(url, { "auth-token": token }).then((response) =>
```
Comment (miromyke) [inline js/chat_app/src/components/DashBoard.js]:
```
function useHttpClient() {
   const auth = useContext(AuthContext)

   return (url, {method, body}) => {
      return fetchJSON(url, {method, body, headers: { 'auth-token': auth.token }})
   }
}

function MyComponent() {
   const fetchJSON = useHttpClient()

   React.useEffect(() => {
      fetchJSON(url).then((response) => {
         console.log(response)
      })
   })
}
```

File: js/chat_app/src/components/ChatList.js [hunk 5]
```diff
@@ -1,48 +1,46 @@
...
-              <MenuItem
-                selected={chatId === params.id}
-                onClick={(event) => handleChatListItemClick(event, params.id)}
-              >
-                {params.title}
-              </MenuItem>
-            );
-          })}
-        </MenuList>
+        {(chatsData.length === 0 && (
```
Comment (miromyke) [inline js/chat_app/src/components/ChatList.js]:
```suggestion
        {(chatsData.length === 0 ? (
```

File: js/chat_app/src/components/ChatList.js [hunk 6]
```diff
@@ -1,48 +1,46 @@
...
+          <Typography align="center" sx={{ p: 1 }}>
+            No existing chats
+          </Typography>
+        )) || (
```
Comment (miromyke) [inline js/chat_app/src/components/ChatList.js]:
```suggestion
        )) : (
```

File: js/chat_app/src/components/ChatList.js [hunk 7]
```diff
@@ -1,48 +1,46 @@
...
+          <MenuList>
+            {chatsData.map((params) => {
```
Comment (miromyke) [inline js/chat_app/src/components/ChatList.js]:
```suggestion
            {chatsData.map((chat) => {
```

File: js/chat_app/src/components/DashBoard.js (hunk 4 above)
Comment (NicolasKul) [reply js/chat_app/src/components/DashBoard.js]: Screenshot 2022-11-03 at 12 30 42 PM
Do you mean something like that?

**Pull request merged**